File: api-example.py
#!/usr/bin/python
# Copyright 2013 Christian Berendt
# Copyright 2013 Luke Dashjr
#
# This program is free software; you can redistribute it and/or modify it under
# the terms of the GNU General Public License as published by the Free Software
# Foundation; either version 3 of the License, or (at your option) any later
# version.  See COPYING for more details.

import argparse
import json
import logging
import pprint
import socket

logging.basicConfig(
         format='%(asctime)s %(levelname)s %(message)s',
         level=logging.DEBUG
)

parser = argparse.ArgumentParser()
parser.add_argument("command", default="summary", nargs='?')
parser.add_argument("parameter", default="", nargs='?')
parser.add_argument("--hostname", default="localhost")
parser.add_argument("--port", type=int, default=4028)
args = parser.parse_args()

s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

try:
    s.connect((args.hostname, args.port))
except socket.error, e:
    logging.error(e)

try:
    s.send("{\"command\" : \"%s\", \"parameter\" : \"%s\"}"
            % (args.command, args.parameter)
          )
except socket.error, e:
    logging.error(e)


data = ''
while True:
    try:
        newdata = s.recv(1024)
        if newdata:
            data += newdata
        else:
            break
    except socket.error, e:
        break

try:
    s.close()
except socket.error,e:
    logging.error(e)

if data:
    data = json.loads(data.replace('\x00', ''))
    pp = pprint.PrettyPrinter()
    pp.pprint(data)

File: api-example.rb
#!/usr/bin/env ruby

# Copyright 2014 James Hilliard
#
# This program is free software; you can redistribute it and/or modify it under
# the terms of the GNU General Public License as published by the Free Software
# Foundation; either version 3 of the License, or (at your option) any later
# version.  See COPYING for more details.

require 'socket'
require 'json'

api_command = ARGV[0].split(":")

if ARGV.length == 3
	api_ip = ARGV[1]
	api_port = ARGV[2]
elsif ARGV.length == 2
	api_ip = ARGV[1]
	api_port = 4028
else
	api_ip = "127.0.0.1"
	api_port = 4028
end

s = TCPSocket.open(api_ip, api_port)

if api_command.count == 2
	s.write({ :command => api_command[0], :parameter => api_command[1]}.to_json)
else
	s.write({ :command => api_command[0]}.to_json)
end

response = s.read.strip
response = JSON.parse(response)

puts response
s.close
